src/NuGet.Core/NuGet.Resolver/CombinationSolver.cs

```diff
@@ -155,22 +155,22 @@ private IEnumerable<T> FindSolution(Action<IEnumerable<T>> diagnosticOutput)
                     // create a new list since this method reuses the solution array
                     diagnosticOutput(new List<T>(_solution));
                 }
+            }
```

Comment:
I dont think this will get the same behavior as before since it might just go into infinite loop when i is moving back and forth. Besides, I dont need think this is a real scenario to pass empty targetIds to resolve api to resolve... what are you really trying to resolve?

Reply:
I'm not sure I follow how it'd be any different than before? I think the infinite loop "risk" is the same it was before this change. 

I'm also curious to hear the real scenario, but I am not against taking this change.

Reply:
Edited for clarity